front/app/components/EventCard/EventInformation.tsx

```diff
@@ -0,0 +1,259 @@
+  margin-right: 6px;
+`;
+
+const EventDescription = styled.div``;
+
+const SMALL_LINE_HEIGHT = fontSizes.small + 2.45;
+
+interface IStyledT {
+  hideTextOverflow?: boolean;
+}
+
+// https://css-tricks.com/line-clampin/#the-fade-out-way
+const StyledT = styled(T)<IStyledT>`
+  ${({ hideTextOverflow }) => {
+    if (!hideTextOverflow) return '';
+
+    return `
+      overflow: hidden;
+      height: calc(${SMALL_LINE_HEIGHT}px * 4);
+
+      &:after {
+        content: "";
+        text-align: right;
+        position: absolute;
+        bottom: 0;
+        right: 0;
+        width: 100%;
+        height: ${SMALL_LINE_HEIGHT * 2}px;
+        background: linear-gradient(to bottom, rgba(255, 255, 255, 0), rgba(255, 255, 255, 1) 100%);
+      }
+    `;
+  }}
+
+  p {
+    font-size: ${fontSizes.small}px;
+    line-height: ${SMALL_LINE_HEIGHT}px;
+  }
+
+  color: ${(props: any) => props.theme.colorText};
+  position: relative;
+  display: block;
+`;
+
+const ReadMoreOrLessWrapper = styled.div`
+  margin-top: 18px;
+`;
+
+const ReadMoreOrLess = styled.a`
+  color: ${colors.label};
+  cursor: pointer;
+  font-weight: 600;
+  text-decoration-line: underline;
+
+  &:hover {
+    color: ${({ theme }) => theme.colorMain};
+    text-decoration-line: underline;
+  }
+`;
+
+interface Props {
+  event: IEventData;
+  startAtMoment: moment.Moment;
+  endAtMoment: moment.Moment;
+  isMultiDayEvent: boolean;
+  showProjectTitle?: boolean;
+}
+
+const EventInformation = memo<Props & InjectedIntlProps>((props) => {
+  const {
+    event,
+    isMultiDayEvent,
+    startAtMoment,
+    endAtMoment,
+    showProjectTitle,
+    intl,
+  } = props;
+
+  const theme: any = useTheme();
+
+  const eventFiles = useResourceFiles({
+    resourceType: 'event',
+    resourceId: event.id,
+  });
+
+  const hasLocation = !every(event.attributes.location_multiloc, isEmpty);
+  const eventDateTime = isMultiDayEvent
+    ? `${startAtMoment.format('LLL')} - ${endAtMoment.format('LLL')}`
+    : `${startAtMoment.format('LL')} • ${startAtMoment.format(
+        'LT'
+      )} - ${endAtMoment.format('LT')}`;
+
+  const projectId = event.relationships.project.data.id;
+  const project = useProject({ projectId });
+  const projectTitle = project?.attributes.title_multiloc;
+
+  const TElement = useRef(null);
+
+  const [textOverflow, setTextOverflow] = useState(true);
+  const [hideTextOverflow, setHideTextOverflow] = useState(true);
+
+  const showHiddenText = () => setHideTextOverflow(false);
+  const hideText = () => setHideTextOverflow(true);
+
+  useEffect(() => {
+    if (textOverflow === false) return;
+
+    setTextOverflow(true);
+
+    setTimeout(() => {
+      setTextOverflow(checkTextOverflow(TElement));
+    }, 0);
+  }, [TElement]);
+
+  return (
+    <>
+      <EventInformationContainer data-testid="EventInformation">
+        <EventTitleAndMeta>
+          {showProjectTitle && projectTitle && (
+            <ProjectTitle>
+              <T value={projectTitle} as="a" />
```

Comment:
Fixed this :+1: it's now not a link but s `span` though, because I added styling and `<T>` with the `as` prop doesn't work with `styled-components` for some reason... Would be nice to have some other way to deal with `MultiLoc`s besides the `<T>` component

Reply:
There is another way: the useLocalize hook or injectLocalize for non-functional components.